### PR TITLE
node: remove dependency on BPF code

### PR DESCRIPTION
**Pull request by ianvernon**
Signed-off by: Ian Vernon <ian@cilium.io>

Related-to: #6101

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6156)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
CI failed to fetch Cilium:

```
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from http://github.com/cilium/cilium
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:888)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1155)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1186)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:144)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:67)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:303)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress http://github.com/cilium/cilium +refs/pull/*:refs/remotes/origin/pr/*" returned status code 128:
```

**Comment by ianvernon** [comment]
test-me-please

File: pkg/node/datapath.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+package node
+
+import "net"
+
+var (
+	// TunnelDatapath is the object which manages the datapath for tunnels.
+	TunnelDatapath TunnelEndpointOwner
+)
+
+// TunnelEndpointOwner is any type which is able to modify tunnel endpoints
```
Comment (joestringer) [inline pkg/node/datapath.go]: Perhaps,

TunnelEndpointOwner configures connectivity for a range of IP addresses (specified by prefix) via a tunnel to a specified remote tunnel endpoint address (node IP).

File: pkg/maps/tunnel/tunnel.go [hunk 2]
```diff
@@ -54,12 +54,18 @@ var (
 			}
 
 			return &k, &v, nil
-		}).WithCache()
+		}).WithCache(),
+	}
 )
 
+// Map is a wrapper around bpf.Map.
```
Comment (joestringer) [inline pkg/maps/tunnel/tunnel.go]: Do you think there's a more useful description we could provide for more context?

I'm thinking something like,

```
Map implements tunnel connectivity configuration in the BPF datapath.
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Nice!

I have a couple of suggestions on the descriptions of objects below.

File: pkg/node/datapath.go (hunk 1 above)
Comment (ianvernon) [reply pkg/node/datapath.go]: Addressed in next upload.

File: pkg/maps/tunnel/tunnel.go (hunk 2 above)
Comment (ianvernon) [reply pkg/maps/tunnel/tunnel.go]: Fixed in next upload.

**Comment by ianvernon** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Pull request merged**